### ISSUE TITLE
popz Empty stack issue.

### ISSUE THREAD

**Issue by Kneesnap** [krzys-h/UndertaleModTool#165]
I wasn't able to fix the bug, and there was no functional consequence in the specific scenario. Makes sense though.

Here's what the code looks like, with the comments:
```
switch dialog_step {
  case 0:
    {
     if (global.enemy_hp_check <= (global.enemy_hp_max * 0.75)) {
      dialog_step = 1
      dialog_active = 1
      cur_message = 14
      text_location = 3
      text_flipped = 1
      alarm[3] = room_speed
      exit
     }
  // Something was wrong with the stack, reentered the block with 1 variables instead of 0, ignoring
  }
  break
  case 1:
    {
     if (global.enemy_hp_check <= (global.enemy_hp_max * 0.5)) {
      dialog_step = 2
      dialog_active = 1
      cur_message = 16
      text_location = 3
      text_flipped = 1
      alarm[3] = room_speed
      exit
     }
  // Something was wrong with the stack, reentered the block with 1 variables instead of 0, ignoring
  }
 // WARNING: Popz'd an empty stack.
 alarm[1] = room_speed
 // Something was wrong with the stack, reentered the block with 1 variables instead of 0, ignoring
 // Something was wrong with the stack, reentered the block with 1 variables instead of 0, ignoring
}
```

Source:
obj_barghest2_controller_Step_0 - Heartbound Demo

I reviewed the byte-code, and I wasn't sure what was going wrong in the decompiler, but I determined that these issues were related and weren't supposed to leave any actual decompiled code behind.

_Originally posted by @Kneesnap in https://github.com/_render_node/MDE3OlB1bGxSZXF1ZXN0UmV2aWV3MjMxNTQxMDA1/pull_request_reviews/more_threads_

**Comment by colinator27** [comment]
I just fixed this issue in my local copy, it has to do with exits/returns and stopping the decompiler and HL decompiler from going down paths that are inaccessible or incorrect.

**Comment by colinator27** [comment]
Fixed in #162
